pkg/preflights/host-preflight.yaml

```diff
@@ -335,35 +335,24 @@ spec:
               when: error
               message: >
                 Error connecting to {{ .ProxyRegistryURL }}.
-                Ensure your firewall is properly configured, or use the `--http-proxy`, `--https-proxy`,
-                and `--no-proxy` flags if there is a proxy server.
+                Ensure your firewall is properly configured, and use the --http-proxy, --https-proxy,
+                and --no-proxy flags if there is a proxy server.
           - pass:
               when: 'statusCode == 401'
               message: 'Connected to {{ .ProxyRegistryURL }}'
           - fail:
               message: >
                 Unexpected response from {{ .ProxyRegistryURL }}.
-                Ensure your firewall is properly configured, or use the `--http-proxy`, `--https-proxy`,
-                and `--no-proxy` flags if there is a proxy server.
-    - textAnalyze:
-        checkName: localhost in resolv.conf
-        fileName: host-collectors/run-host/resolv.conf.txt
-        regex: 'nameserver localhost'
-        outcomes:
-          - fail:
-              when: "true"
-              message: "resolv.conf should not reference 'localhost', this will prevent coredns within the cluster from resolving"
-          - pass:
-              when: "false"
-              message: "resolv.conf does not reference 'localhost'"
+                Ensure your firewall is properly configured, and use the --http-proxy, --https-proxy,
+                and --no-proxy flags if there is a proxy server.
     - textAnalyze:
-        checkName: 127.0.0.1 in resolv.conf
+        checkName: Resolver Configuration
         fileName: host-collectors/run-host/resolv.conf.txt
-        regex: 'nameserver 127.0.0.1'
+        regex: 'nameserver\s*(localhost|127\.0\.0\.1)'
```

Comment:
Made this a single preflight so someone wouldn't remove `nameserver localhost` and try `nameserver 127.0.0.1` and fail again.